src/HTTP/Cookies/CookieWrapper.php

```diff
@@ -0,0 +1,265 @@
+<?php
+
+namespace ILIAS\HTTP\Cookies;
+
+use Dflydev\FigCookies\SetCookie;
+
+/**
+ * Class CookieFacade
+ *
+ * Facade class for the FigCookies SetCookie class.
+ *
+ * @author  Nicolas Schäfli <ns@studer-raimann.ch>
+ * @package ILIAS\HTTP\Cookies
+ * @since   5.2
+ * @version 1.0.0
+ */
+class CookieWrapper implements Cookie {
+
```

Comment:
@Amstutz 
I made this comment about 11 days ago. Since that d3r1w changed the implementation in immutable. So my comments are obsolete.

You're right. Looks like _clone_ is a little cheaper then _new_. I don't know how often it will be called in this case, but i think it could be negligible. Benchmark runs i found shows differences about 0,2 seconds if cloning 4 object in 100.000 cycles.